src/memorylayout.jl

```diff
@@ -257,6 +283,65 @@ transposelayout(::ConjLayout{ML}) where ML = ConjLayout{typeof(transposelayout(M
 adjointlayout(::Type{T}, M::MemoryLayout) where T = transposelayout(conjlayout(T, M))
 
 
+# Layouts of PermutedDimsArrays
+"""
+    UnitStride{D}()
+
+is returned by `MemoryLayout(A)` for arrays of ndims(A) >= 3 for which `stride(A,D) == 1`.
+
+`UnitStride{1}` is weaker than `ColumnMajor` in that it does not demand that the other
+strides are increasing, hence is is not a subtype of `AbstractIncreasingStrides`.
+To ensure that `stride(A,D) == 1`, you may dispatch on `Union{UnitStride{1}, AbstractColumnMajor}`
+to allow for both options. (With complex numbers, you may also need their `ConjLayout` versions.)
+
+Likewise, both `UnitStride{ndims(A)}` and `AbstractRowMajor` have `stride(A, ndims(A)) == 1`.
+"""
+UnitStride
+
+MemoryLayout(::Type{PermutedDimsArray{T,N,P,Q,S}}) where {T,N,P,Q,S} = permutelayout(MemoryLayout(S), Val(P))
+
+permutelayout(::Any, perm) = UnknownLayout()
+permutelayout(::StridedLayout, perm) = StridedLayout()
+permutelayout(::ConjLayout{ML}, perm) where ML = ConjLayout{typeof(permutelayout(ML(), perm))}()
+
+function permutelayout(layout::AbstractColumnMajor, ::Val{perm}) where {perm}
+    issorted(perm) && return layout
+    issorted(reverse(perm)) && return reverse(layout)
+    tup = ntuple(length(perm)) do D
+        perm[D] == 1 && return UnitStride{D}()
+    end
+    only(filter(x -> x isa MemoryLayout, tup))
```

Comment:
`filter` needs Compat for only Julia versions... perhaps there's another way to do this?

Otherwise it's just a few lines from https://github.com/JuliaLang/Compat.jl/pull/682/files which could be copied instead.